CSharpMath/Generic/BiDictionary.cs

```diff
@@ -9,8 +9,8 @@ namespace CSharpMath
 {
   public class AliasDictionary<K, V> : ICollection<KeyValuePair<K, V>>, IDictionary<K, V> {
     public AliasDictionary() {
-      k2v = new Dictionary<K, V>();
-      v2k = new Dictionary<V, K>();
+      k2v = new Dictionary<K, V>(293);
+      v2k = new Dictionary<V, K>(293);
```

Comment:
293 seems arbitrary. More commands can and will be defined so this number is going to be forgotten thus will not be updated. Moreover, this is a general purpose type and can be used in other places so it will be better to follow the .NET convention of not occupying much space until needed. You can use a constructor that takes an initial capacity though.

Reply:
Actually there was adding 247 items if i'm not wrong, 293 just a next prime number after 247, which hashtable uses for next array where previous array will be copied and reallocated. Believe me this is much much more better  then adding items until 247. If you don't wont a bit more space, we can make capacity 247, but then once we add another one next array size will be 293 and reallocated, thats why i've chosen 293

Reply:
Here it explained https://www.codeproject.com/Articles/500644/Understanding-Generic-Dictionary-in-depth

Reply:
This is a one-time allocation only so the performance improvement will be limited.

Reply:
Also, this is a Dictionary type. You supply the initial capacity in the constructor, not embedded directly in the type itself. 

E.g. `new AliasDictionary<,>(297)` instead of defaulting to 297

Reply:
So what i do? keep capacity or delete ? i'm going to do fixes on my PR

Reply:
You should probably add another constructor like the built-in `Dictionary` type:
```cs
    public AliasDictionary(int initialCapacity) {
      k2v = new Dictionary<K, V>(initialCapacity);
      v2k = new Dictionary<V, K>(initialCapacity);
    }
```